hashing/te-tag-query-python/all_in_one_lib/signal_type/base.py

```diff
@@ -0,0 +1,156 @@
+##!/usr/bin/env python
+
+"""
+Abstraction for different signal types.
+
+This helps translates ThreatDescriptors into the correct representation to do
+matching, as well as serialize that representation into a compact form.
+"""
+
+import csv
+import pathlib
+import re
+import typing as t
+
+from .. import common
+from ..descriptor import SimpleDescriptorRollup
+
+
+class SignalMatch(t.NamedTuple):
+    # TODO - Labels probably don't belong here, because we just duplicate storage
+    #        better to have descriptor lookup with the full labels
+    labels: t.Set[str]
+    primary_descriptor_id: int
+    # TODO: Someday, also include the following:
+    # contested: bool
+    # plurality_opinion: t.Tuple(bool, t.Set[str])
+    # highest_action_severity: str
+    # opinions_by_owner: t.Dict[int, MatchOpinion]
```

Comment:
Note to self, I actually concluded that I probably want to separate the descriptor: label/opinion storage from the match storage, since I otherwise store the same information on multiple signal type implementations. Add a comment and delete this junk